datumaro/plugins/pointcloud_format/extractor.py

```diff
@@ -0,0 +1,131 @@
+import json
+import os
+
+from collections import OrderedDict
+import os.path as osp
+
+
+from datumaro.components.extractor import (SourceExtractor, DatasetItem,
+    AnnotationType,Cuboid,
+    LabelCategories, Importer
+)
+
+from datumaro.util.image import Image
+
+from .format import PointCloudPath
+
+
+class PointCloudExtractor(SourceExtractor):
+    _SUPPORTED_SHAPES = "cuboid"
+    mapping = {}
+    meta = {}
+
+    def __init__(self, path, subset=None):
+        assert osp.isfile(path), path
+        rootpath = osp.dirname(path)
+        images_dir = ''
+        if osp.isdir(osp.join(rootpath, PointCloudPath.ANNNOTATION_DIR)):
+            images_dir = osp.join(rootpath, PointCloudPath.ANNNOTATION_DIR)
+        self._images_dir = images_dir
+        self._path = path
+
+        if not subset:
+            subset = osp.splitext(osp.basename(path))[0]
+
+        super().__init__(subset=subset)
+
+        items, categories = self._parse(path)
+        self._items = list(self._load_items(items).values())
+        self._categories = categories
+
+    @classmethod
+    def _parse(cls, path):
+        path = osp.abspath(path)
+        items = OrderedDict()
+        categories = {}
+
+        if osp.split(path)[-1] == "key_id_map.json":
+            with open(path, "r") as f:
+                data = f.read()
+            cls.mapping = json.loads(data)
+
+        meta_path = osp.abspath(osp.join(osp.dirname(path), "meta.json"))
+
+        if osp.isfile(meta_path):
+
+            with open(meta_path, "r") as f:
+                meta = f.read()
+
+            cls.meta = json.loads(meta)
+
+        label_cat = LabelCategories()
+        common_attrs = ["occluded"]
+
+        if cls.meta:
+            for label in cls.meta["classes"]:
+                label_cat.add(label['title'], attributes=common_attrs)
```

Comment:
```suggestion
                label_cat.add(label['title'])
```
Keep only label-specific attributes here, the common ones are supposed to be in `LabelCategories`

Reply:
Resolved